crates/solidity/src/process/mod.rs

```diff
@@ -17,13 +17,21 @@ use self::output::Output;
 pub static EXECUTABLE: OnceCell<PathBuf> = OnceCell::new();
 
 /// Read input from `stdin`, compile a contract, and write the output to `stdout`.
-pub fn run() -> anyhow::Result<()> {
-    let mut stdin = std::io::stdin();
+pub fn run(input_file: Option<&mut std::fs::File>) -> anyhow::Result<()> {
     let mut stdout = std::io::stdout();
     let mut stderr = std::io::stderr();
+    let mut stdin = std::io::stdin();
 
     let mut buffer = Vec::with_capacity(16384);
-    stdin.read_to_end(&mut buffer).expect("Stdin reading error");
+    match input_file {
+        Some(ins) => {
+            ins.read_to_end(&mut buffer)
+                .expect("Error reading from input file");
```

Comment:
Nit: We usually unwrap results in case we are guaranteed to never panic (the rust compiler can not now) and use `expect(...)` to explain the reason as to why the unwrap would never panic.

But since the this function does already return a `anhow::Result` it seems like here we can be just bubble up the error instead?